src/test/java/core/basesyntax/strategy/OperationStrategyTest.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.strategy;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.impl.BalanceOperationHandler;
+import core.basesyntax.strategy.impl.PurchaseOperationHandler;
+import core.basesyntax.strategy.impl.ReturnOperationHandler;
+import core.basesyntax.strategy.impl.SupplyOperationHandler;
+import org.junit.Test;
+
+public class OperationStrategyTest {
+    private static final OperationStrategy operationStrategy = new OperationStrategy();
+    private static final FruitTransaction fruitTransaction = new FruitTransaction(
+            FruitTransaction.Operation.BALANCE, "apple", 100);
+
+    @Test
+    public void getHandler_validTransactions_ok() {
+        fruitTransaction.setOperation(FruitTransaction.Operation.BALANCE);
+        assertEquals("invalid return class", operationStrategy
+                .getHandler(fruitTransaction).getClass(), BalanceOperationHandler.class);
+        fruitTransaction.setOperation(FruitTransaction.Operation.PURCHASE);
+        assertEquals("invalid return class", operationStrategy
+                .getHandler(fruitTransaction).getClass(), PurchaseOperationHandler.class);
+        fruitTransaction.setOperation(FruitTransaction.Operation.SUPPLY);
+        assertEquals("invalid return class", operationStrategy
+                .getHandler(fruitTransaction).getClass(), SupplyOperationHandler.class);
+        fruitTransaction.setOperation(FruitTransaction.Operation.RETURN);
+        assertEquals("invalid return class", operationStrategy
+                .getHandler(fruitTransaction).getClass(), ReturnOperationHandler.class);
+    }
+
+    @Test(expected = RuntimeException.class)
```

Comment:
I still think its better to create custom exception, ex: OperationHandlerException as it will help us in debugging when project becomes really big